tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,222 @@
+					GinkgoWriter.Println("Istio CR created")
+
+					resourceReconcilied = resourcecondition.Conditions{
+						Type:   "Reconciled",
+						Status: "True",
+					}
+
+					resourceReady = resourcecondition.Conditions{
+						Type:   "Ready",
+						Status: "True",
+					}
+				})
+
+				AfterAll(func() {
+					Eventually(kubectl.DeleteNamespace).WithArguments(controlPlaneNamespace).Should(Succeed(), "Namespace should be deleted")
+					Eventually(kubectl.GetNamespace).WithArguments(controlPlaneNamespace).Should(Equal(fmt.Sprintf("namespace %s not found", controlPlaneNamespace)))
+					GinkgoWriter.Println("Cleanup done")
+				})
+
+				When("the Istio resource is created", func() {
+					It("updates the Istio resource status to Ready and Running", func() {
+						Eventually(func(g Gomega) {
+							g.Expect(kubectl.GetResourceCondition(controlPlaneNamespace, "istio", istioName)).Should(ContainElement(resourceReconcilied))
+						}).Should(Succeed(), "Istio CR should be reconciled")
+
+						Eventually(func(g Gomega) {
+							g.Expect(kubectl.GetResourceCondition(controlPlaneNamespace, "istio", istioName)).Should(ContainElement(resourceReady))
+						}).Should(Succeed(), "Istio CR should be Ready")
+
+						istiodPodName, _ := kubectl.GetPodFromLabel(controlPlaneNamespace, "app=istiod")
+						GinkgoWriter.Println("Istiod pod name: ", istiodPodName)
+
+						Eventually(kubectl.GetPodPhase).WithArguments(controlPlaneNamespace, istiodPodName).Should(Equal("Running"), "Istiod should be Running")
+					})
+
+					Specify("the istio resource version match the applied version", func() {
+						var istio v1alpha1.Istio
+						output, _ := kubectl.GetResource(controlPlaneNamespace, "istio", istioName)
+
+						err := json.Unmarshal([]byte(output), &istio)
+						if err != nil {
+							Fail("Error unmarshalling the IstioCr")
+						}
+
+						Expect(istio.Spec.Version).To(Equal(version), "Istio CR version should match the applied version")
+					})
+
+					It("istio resource stopped reconciling", func() {
+						istiodPodName, _ := kubectl.GetPodFromLabel(controlPlaneNamespace, "app=istiod")
+						Eventually(kubectl.GetPodLogs).WithArguments(controlPlaneNamespace, istiodPodName, "30s").ShouldNot(ContainSubstring("Reconciliation done"))
+						GinkgoWriter.Println("Istio Operator stopped reconciling")
+					})
+
+					It("only istiod is deployed", func() {
+						expectedDeployments := []string{"istiod"}
+
+						deploymentsList, err := kubectl.GetDeployments(controlPlaneNamespace)
+						if err != nil {
+							Fail("Error getting deployments")
+						}
+
+						Expect(deploymentsList).To(Equal(expectedDeployments), "Deployments List should contains only istiod")
+					})
+
+					It("deploys the CNI DaemonSet when running on OpenShift", func() {
+						daemonsetsList, err := kubectl.GetDaemonSets(namespace)
+						if err != nil {
+							Fail(fmt.Sprintf("Error getting daemonsets: %s", err))
+						}
```

Comment:
I don't think it makes much sense to have this variable. I know that you're using it in two places, but because you have to handle errors when you introduce this variable, I think it makes more sense to simply duplicate the function call in order to take advantage of Ginkgo's implicit error handling.

Reply:
Done